Bridge/Symfony/Routing/ApiLoader.php

```diff
@@ -0,0 +1,154 @@
+     */
+    private $propertyItemMetadataFactory;
+
+    public function __construct(KernelInterface $kernel, ResourceCollectionMetadataFactoryInterface $resourceCollectionMetadataFactory, ResourceItemMetadataFactoryInterface $resourceItemMetadataFactory, PropertyItemMetadataFactoryInterface $propertyItemMetadataFactory)
+    {
+        $this->fileLoader = new XmlFileLoader(new FileLocator($kernel->locateResource('@DunglasApiBundle/Resources/config/routing')));
+        $this->resourceCollectionMetadataFactory = $resourceCollectionMetadataFactory;
+        $this->resourceItemMetadataFactory = $resourceItemMetadataFactory;
+        $this->propertyItemMetadataFactory = $propertyItemMetadataFactory;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function load($data, $type = null)
+    {
+        $routeCollection = new RouteCollection();
+
+        $routeCollection->addCollection($this->fileLoader->load('jsonld.xml'));
+        $routeCollection->addCollection($this->fileLoader->load('hydra.xml'));
+
+        foreach ($this->resourceCollectionMetadataFactory->create() as $resourceClass) {
+            $itemMetadata = $this->resourceItemMetadataFactory->create($resourceClass);
+            $normalizedShortName = Inflector::pluralize(Inflector::tableize($itemMetadata->getShortName()));
+
+            foreach ($itemMetadata->getCollectionOperations() as $operationName => $operation) {
+                $this->addRoute($routeCollection, $resourceClass, $operationName, $operation, $normalizedShortName, true);
+            }
+
+            foreach ($itemMetadata->getItemOperations() as $operationName => $operation) {
+                $this->addRoute($routeCollection, $resourceClass, $operationName, $operation, $normalizedShortName, false);
+            }
+        }
+
+        return $routeCollection;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supports($resource, $type = null)
+    {
+        return 'api' === $type;
+    }
+
+    /**
+     * Creates and adds a route for the given operation to the route collection.
+     *
+     * @param RouteCollection $routeCollection
+     * @param string          $resourceClass
+     * @param string          $operationName
+     * @param Operation       $operation
+     * @param string          $normalizedShortName
+     * @param bool            $collection
+     *
+     * @throws RuntimeException
+     */
+    private function addRoute(RouteCollection $routeCollection, string $resourceClass, string $operationName, Operation $operation, string $normalizedShortName, bool $collection)
+    {
+        $attributes = $operation->getAttributes();
+
+        if (isset($attributes['route_name'])) {
+            return;
+        }
+
+        if (!isset($attributes['method'])) {
+            throw new RuntimeException('Either a "route_name" or a "method" operation attribute must exist.');
+        }
+
+        if (isset($attributes['controller'])) {
```

Comment:
I guess this needs to be a negative? Because with the `else` you would get an undefined key notice.